yt_dlp/extractor/youtube/_video.py

```diff
@@ -3950,10 +4017,30 @@ def process_language(container, base_url, lang_code, sub_name, query):
                         # Add an "-orig" label to the original language so that it can be distinguished.
                         # The subs are returned without "-orig" as well for compatibility
                         process_language(
-                            automatic_captions, base_url, f'{trans_code}-orig', f'{trans_name} (Original)', {})
+                            automatic_captions, base_url, f'{trans_code}-orig',
+                            f'{trans_name} (Original)', client_name, pot_params)
                     # Setting tlang=lang returns damaged subtitles.
-                    process_language(automatic_captions, base_url, trans_code, trans_name,
-                                     {} if orig_lang == orig_trans_code else {'tlang': trans_code})
+                    process_language(
+                        automatic_captions, base_url, trans_code, trans_name, client_name,
+                        pot_params if orig_lang == orig_trans_code else {'tlang': trans_code, **pot_params})
+
+            # Avoid duplication if we got everything we needed from the first player reponse
+            need_subs_langs.difference_update(subtitles)
+            need_caps_langs.difference_update(automatic_captions)
+            if not (need_subs_langs or need_caps_langs):
+                break
+
+        subs_wanted = any((
+            self.get_param('writesubtitles'),
+            self.get_param('writeautomaticsub'),
+            self.get_param('listsubtitles')))
+        if subs_wanted and skipped_subs_clients and (need_subs_langs or need_caps_langs):
+            self.report_warning(join_nonempty(
+                'One or more clients are requiring PO tokens for access to',
+                f'subtitles and automatic captions ({", ".join(skipped_subs_clients)}).',
+                need_subs_langs and f'Subtitles for these languages are missing: {", ".join(need_subs_langs)}.',
+                need_caps_langs and f'Automatic captions for these languages are missing: {", ".join(need_caps_langs)}',
+                delim=' '), video_id=video_id)
```

Comment:
Originally I liked the idea of showing the user what exactly they're missing, but now I realize that if **all** clients are skipped, then the automatic captions list could be like a full page warning. We could keep the sub langs list, but for captions just say "Some automatic captions are missing" ?

Reply:
I think that works. Or if the size of the captions list is too long, simplify it to a number like "42 subtitle langs were skipped" perhaps?